### ISSUE TITLE
The true_r in predictions turns negative value 

### ISSUE THREAD

**Issue by fjssharpsword** [NicolasHug/Surprise#286]
#### Description

In my examples, the rating_scale is (0, 1).  After executing "predictions = algo.test(testset)",  I find that the true_r in "predictions" turns to negative value.
For example:
(u,i,0.05)-->(u,i,-0.95)
In this case , it is wrong to calculate the error between ture_r and est.
why?

**Comment by NicolasHug** [comment]
World you have a minimal reproducible example?

**Comment by fjssharpsword** [comment]
`#SVD，Surprise
import surprise as sp
import pandas as pd
import numpy as np

#1.Loading the dataset and Excluding the outliers
kbdata = pd.read_csv("/data/fjsdata/ctKngBase/kb.csv", sep='|', low_memory=False)
kbdata = kbdata.loc[(kbdata['num']<200)]#7个月，一天一次估算
num_max=kbdata['num'].max()
num_min=kbdata['num'].min()
kbdata['num']=kbdata['num'].apply(lambda x: (x-num_min+1)*1.0/(num_max-num_min+1) )
print ('Dataset shape is:%d rows and %d columns'%(kbdata.shape[0],kbdata.shape[1]))
testset = kbdata.sample(frac=0.1)
#2.Transforming into data format of surprise and spliting the train-set and test-set
# The columns must correspond to user id, item id and ratings (in that order).
reader = sp.Reader(rating_scale=(0, 1))
spdata = sp.Dataset.load_from_df(kbdata[['csr', 'ke', 'num']],reader)
trainset = spdata.build_full_trainset()
testset = np.array(testset).tolist()
# sampling random trainset and testset, and test set is made of 10% of the ratings.
#trainset, testset = sp.model_selection.train_test_split(spdata, test_size=.1)

#3.Training the model
algo = sp.SVD()
algo.fit(trainset)

#4.predicting ratings for the testset
predictions = algo.test(testset)
for uid, iid, true_r, est, _ in predictions:
    print (true_r)
    break;
print (sp.accuracy.rmse(predictions))`

Dataset shape is:2547452 rows and 3 columns
**-0.7939698492462312**
RMSE: 0.9983
0.9983334614928954

**Comment by fjssharpsword** [comment]
By debugging, I find the problem is "testset = np.array(testset).tolist()",  where I turn dataframe to list.
If I use "model_selection.train_test_split(spdata, test_size=.1)", it is normal. But if I split the testset by dataframe.sample, then turn dataframe to list, it is wrong.

**Comment by fjssharpsword** [comment]
In summary, when I use "testset = np.array(testset).tolist()",  in which I turn dataframe to list, Surprise will return negative value of true_r after "predictions = algo.test(testset)".  
e.g.
ture_r = 0.2060, after prediction, will be (0.2060-1)=-.7940, where the rating scale is (0,1).

**Comment by NicolasHug** [comment]
You should not split your data before creating the dataset object.

**Comment by fjssharpsword** [comment]
thanks!
In my view, this isn't  an effective solution.
